src/renditions.rs

```diff
@@ -0,0 +1,186 @@
+use super::*;
+
+/// Writer for an _rendition dictionary_.
+///
+/// This struct is created by [`Action::rendition`].
+pub struct Rendition<'a> {
+    dict: Dict<'a>,
+}
+
+writer!(Rendition: |obj| {
+    let mut dict = obj.dict();
+    dict.pair(Name(b"Type"), Name(b"Rendition"));
+    Self { dict }
+});
+
+impl<'a> Rendition<'a> {
+    /// Write the `/S` attribute to set the rendition type.
+    pub fn rendition_type(&mut self, kind: RenditionType) -> &mut Self {
+        self.pair(Name(b"S"), kind.to_name());
+        self
+    }
+
+    /// Start writing the `/C`, i.e. media clip, dictionary which specifies what
+    /// media should be played. Only permissible for Media Renditions.
+    pub fn media_clip(&mut self) -> MediaClip<'_> {
+        self.insert(Name(b"C")).start()
+    }
+
+    /// Start writing the `/P`, i.e. media play parameters, dictionary which
+    /// specifies how the media should be played. Only permissible for Media
+    /// Renditions.
+    pub fn media_play_params(&mut self) -> MediaPlayParams<'_> {
+        self.insert(Name(b"P")).start()
+    }
+}
+
+deref!('a, Rendition<'a> => Dict<'a>, dict);
+
+/// Writer for an _media clip dictionary_.
+///
+/// This struct is created by [`Rendition::media_clip`].
+pub struct MediaClip<'a> {
+    dict: Dict<'a>,
+}
+
+writer!(MediaClip: |obj| {
+    let mut dict = obj.dict();
+    dict.pair(Name(b"Type"), Name(b"MediaClip"));
+    Self { dict }
+});
+
+impl<'a> MediaClip<'a> {
+    /// Write the `/S` attribute to set the media clip type.
+    pub fn media_clip_type(&mut self, kind: MediaClipType) -> &mut Self {
+        self.pair(Name(b"S"), kind.to_name());
+        self
+    }
+
+    /// Start writing the `/D` dictionary specifying the media data.
+    pub fn data(&mut self) -> FileSpec<'_> {
+        self.insert(Name(b"D")).start()
+    }
+
+    /// Writing the `/D` dictionary including the file specification as a link,
+    /// i.e. an URL, to a given path.
+    pub fn data_url(&mut self, path: Str) -> &mut Self {
+        self.data()
+            .file_system(Name(b"URL"))
+            .path(path);
+        self
+    }
+
+    /// Writing the `/D` dictionary including the file specification as
+    /// embedded file referenced by a given id.
+    pub fn data_embedded(&mut self, id: Ref) -> &mut Self {
+        self.data()
+            .path(Str(b"<embedded file>"))
+            .embedded_file(id);
+        self
+    }
```

Comment:
We think that these `impl`s are too opinionated / high-level for `pdf-writer` and that the user can just use `MediaClip::Data`

Reply:
I added those two functions as they are the only sane option for this and avoid the user having to deal with some technical details. There are other instances in this library which implement similar functionality. It would thus be only consistent to keep them, but I don't have a strong opinion on that. I have removed them now.